### PR TITLE
[Bug][seatunnel-connector-spark-tidb] Spark tidb Connector Hadoop dependency excluded

### PR DESCRIPTION
**Pull request by Carl-Zhou-CN**
## Check list

* [ ] Code changed are covered with tests, or it does not need tests for reason:
* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs

### CODE REVIEW

**Comment by Carl-Zhou-CN** [comment]
#2698

**Comment by Carl-Zhou-CN** [comment]
@hailin0 please help to review it

**Comment by hailin0** [comment]
Can you add e2e-testcase(spark)?

reference
https://github.com/apache/incubator-seatunnel/tree/dev/seatunnel-e2e/seatunnel-spark-e2e

**Comment by Carl-Zhou-CN** [comment]
@hailin0 I didn't find anything about TiDB in E2E, do I need to add a module for this? If so, I can try, and how can this PR review be approved to begin

**Comment by Hisoka-X** [comment]
Hi, Please fix license check and codestyle. Thanks!

File: seatunnel-e2e/seatunnel-spark-e2e/seatunnel-connector-spark-tidb-e2e/src/test/resources/tidb/tidb_source_and_sink.conf [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+}
+
+transform {
+
+}
+
+sink {
+
+    tidb {
+        addr = "127.0.0.1"
```
Comment (hailin0) [inline seatunnel-e2e/seatunnel-spark-e2e/seatunnel-connector-spark-tidb-e2e/src/test/resources/tidb/tidb_source_and_sink.conf]: use `addr  = "spark_e2e_tidb"`?

Comment (Carl-Zhou-CN) [reply seatunnel-e2e/seatunnel-spark-e2e/seatunnel-connector-spark-tidb-e2e/src/test/resources/tidb/tidb_source_and_sink.conf]: Yes, I expect so. But now I come across a problem. It seems that I can't deploy tidb in a single docker and connect to it in the way of tispark. It lacks pd and tikv. Do you have any good suggestions?

Comment (hailin0) [reply seatunnel-e2e/seatunnel-spark-e2e/seatunnel-connector-spark-tidb-e2e/src/test/resources/tidb/tidb_source_and_sink.conf]: You can use the container to start all dependencies

@Carl-Zhou-CN 

reference
https://github.com/apache/incubator-seatunnel/pull/2830/files#diff-813f5496f69f3fe1d81254c15504b8c5ea36ec8f9fb2bd90c600429de401a3daR72

**Comment by Carl-Zhou-CN** [comment]
@hailin0 @Hisoka-X I'm sorry that I tried many methods to make the connection between pd and tidb impossible. I also tried to use the e2e samples of tikv contributors, but the same problem still exists. I may not be able to implement the e2e of this module, but I think my submission is meaningful for this connector

**Comment by CalvinKirs** [comment]
hi, thanks for your contribution, dev branch will no longer maintain the V1 version of the connector, so I will close this PR,
If you are interested in this, you can contribute the corresponding V2Connector
https://github.com/apache/incubator-seatunnel/tree/dev/seatunnel-connectors-v2

**Pull request closed without merging**